### PR TITLE
Test minimal windows build using Github actions

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ilammy/msvc-dev-cmd@v1
+    - name: configbuildandtest
+      env:
+        CONFIG_OPTS: no-bulk no-deprecated no-fips no-asm -DOPENSSL_SMALL_FOOTPRINT
+        TARGET: VC-WIN64A
+      run: |
+        New-Item -Path . -Name "_build" -ItemType "directory"
+        Set-Location "_build"
```
Comment (levitte) [inline .github/workflows/windows.yml:17]: I didn't know that we have powershell access like that. That being said, what's wrong with `mkdir` and `cd`?

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        perl ..\Configure no-makedepend no-bulk no-deprecated no-fips no-asm -DOPENSSL_SMALL_FOOTPRINT VC-WIN64A
+        perl configdata.pm --dump
+        nmake build_all_generated
+        nmake PERL=no-perl
+        cmd /c nmake test VERBOSE_FAILURE=yes TESTS=-test_fuzz 2>&1
```
Comment (levitte) [inline .github/workflows/windows.yml:22]: Why the explicit invokation of cmd? Also, I don't think that redirection is needed here...

File: .github/workflows/windows.yml (hunk 1 above)
Comment (slontis) [reply .github/workflows/windows.yml:17]: It says it is a powershell prompt so I just used those commands.. I was having some unrelated issues - will change it back when it works end to end.
Do I need to do the "nmake build_all_generated" ??
Also the tests abruptly failed and I have no idea why yet - does that cmd /c line look ok to you? maybe it timed out??

**Comment by levitte** [comment]
Might í suggest splitting up the tasks in several steps instead of one single run?

**Comment by slontis** [comment]
> Might í suggest splitting up the tasks in several steps instead of one single run?

I was doing this because the ilammy/msvc-dev-cmd@v1 does the MSDEV command line prompt.. I wasnt sure how persistent things were..

**Comment by slontis** [comment]
Woohoo - minimal windows build and test in 13 minutes..

File: .github/workflows/windows.yml (hunk 1 above)
Comment (levitte) [reply .github/workflows/windows.yml:14]: > Do I need to do the "nmake build_all_generated" ??

Not really...  we should maintain a test of that *somewhere* though.  But that should be a separate CI job.
However, when you remove this, remember to change `nmake PERL=no-perl` to just `nmake`

Comment (levitte) [reply .github/workflows/windows.yml:14]: > Also the tests abruptly failed and I have no idea why yet - does that cmd /c line look ok to you? maybe it timed out??

That line looks perfectly fine, and it should be perfectly safe, I hope, to just drop `cmd /c`...  (its presence in appveyor.yml has its explanation, here's not the place to dive into that, though)

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    - name: build
+      run: |
+        cd _build
+        nmake
+        cd ..
+    - name: test
+      run: |
+        cd _build
+        nmake test VERBOSE_FAILURE=yes TESTS=-test_fuzz
+        cd ..
```
Comment (levitte) [inline .github/workflows/windows.yml:27]: Something I would suggest is to have a separate preparation task to create the build directory, then use `working-directory:` with the remaining tasks...  so, something like this:

``` yaml
    steps:
    - uses: actions/checkout@v2
    - uses: ilammy/msvc-dev-cmd@v1
    - name: prepare the build directory
      run: mkdir _build
    - name: config
      working-directory: _build
      run: |
        perl ..\Configure no-makedepend no-bulk no-deprecated no-fips no-asm -DOPENSSL_SMALL_FOOTPRINT VC-WIN64A
        perl configdata.pm --dump
    - name: build
      working-directory: _build
      run: nmake
    - name: test
      working-directory: _build
      run: nmake test VERBOSE_FAILURE=yes TESTS=-test_fuzz
 ```

Comment (slontis) [reply .github/workflows/windows.yml:27]: yep - good idea..

**Comment by slontis** [comment]
This is quite readable now.

**Review by levitte — APPROVED** [review]

**Comment by slontis** [comment]
Thanks for the improvements @levitte. Merged to master.

**Pull request closed without merging**